### PR TITLE
[IMPROVEMENT] Migrate from githubpy to PyGithub

### PR DESCRIPTION
**Pull request by Tarun-Arora**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---

This pull request migrates the old code base from githubpy to PyGithub as githubpy is no longer being maintained
This PR is a part of #761.

### CODE REVIEW

**Comment by Tarun-Arora** [comment]
`nose` doesn't support python 3.10. I will add python 3.10 in the python versions list for GitHub actions while migrating from `nose` to `nose2`

File: mod_ci/controllers.py [hunk 1]
```diff
@@ -1005,27 +993,33 @@ def start_ci():
...
-                github_status = repository.statuses(commit_hash)
+                github_status = repository.get_commit(commit_hash)
 
                 if payload['action'] == "completed":
                     is_complete = True
                     has_failed = False
                     builds = {"linux": False, "windows": False}
-                    for workflow in repository.actions.runs.get(
+
+                    # NOTE: Using this workaround because of bug in PyGithub
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py:1003]: Is this an open bug? Better refer to it then, so we can easily check in the future if it's fixed or not.

Comment (canihavesomecoffee) [reply mod_ci/controllers.py:1003]: If not reported yet, have you reported it?

Comment (Tarun-Arora) [reply mod_ci/controllers.py:1003]: I checked the bug is not reported yet; I'll create an issue there and reference it here
Also, I am yet to write unit tests for this PR.

Comment (Tarun-Arora) [reply mod_ci/controllers.py]: @canihavesomecoffee @thealphadollar 
The issue already existed: https://github.com/PyGithub/PyGithub/issues/2276; I have also referenced this in the code.
The unit tests for this PR are also complete now.

**Review by canihavesomecoffee — APPROVED** [review]

**Review by thealphadollar — APPROVED** [review]
Thanks for the changes, @Tarun-Arora 

The tests also look good to me.

**Comment by thealphadollar** [comment]
@canihavesomecoffee When do we want to merge the PR?

**Comment by canihavesomecoffee** [comment]
Now :partying_face:

**Pull request merged**